### PR TITLE
Unshallow clone

### LINKED ISSUE #84: new project created with CLI, unable to push to new repo
```
pi@raspberrypi:~/nio/projects $ nio new aiy_kit
...
(do some stuff, make some commits)
...
pi@raspberrypi:~/nio/projects/aiy_kit $ git status
On branch master
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   nio.conf

no changes added to commit (use "git add" and/or "git commit -a")
pi@raspberrypi:~/nio/projects/aiy_kit $ git remote -v
pi@raspberrypi:~/nio/projects/aiy_kit $ git remote add origin git@github.com:tyoungNIO/aiy_kit.git
pi@raspberrypi:~/nio/projects/aiy_kit $ git push --set-upstream origin master
Counting objects: 32, done.
Compressing objects: 100% (26/26), done.
Writing objects: 100% (32/32), 12.66 KiB | 0 bytes/s, done.
Total 32 (delta 5), reused 20 (delta 0)
remote: Resolving deltas: 100% (5/5), done.
To github.com:tyoungNIO/aiy_kit.git
 ! [remote rejected] master -> master (shallow update not allowed)
error: failed to push some refs to 'git@github.com:tyoungNIO/aiy_kit.git'
pi@raspberrypi:~/nio/projects/aiy_kit $ 
```

top answer to this question seems to address it:
https://stackoverflow.com/questions/28983842/remote-rejected-shallow-update-not-allowed-after-changing-git-remote-url
`new` and `newblock` commands include a `depth` argument, is this working as intended?
https://github.com/niolabs/nio-cli/search?utf8=%E2%9C%93&q=--depth&type=

### PR DESCRIPTION
**Pull request by tyoungNIO**
close #84 

## Description
Any additional info regarding implementation and how to replicate locally

## JIRA Issue (Optional)
[NIO-xyz](https://neutralio.atlassian.net/browse/NIO-xyz)

## Todos
- [x] Tested and working locally
- [x] Unit Tests
- [ ] Updated [documentation](https://github.com/niolabs/docs)
- [x] Add a label to the PR

### CODE REVIEW

File: nio_cli/commands/new.py [hunk 1]
```diff
@@ -27,15 +27,15 @@ def run(self):
...
-        clone = "git clone --depth=1 {} {}".format(repo, self._name)
+        clone = "git clone {} {}".format(repo, self._name)
         submodule_update = (
             'cd ./{} '
             '&& git submodule update --init --recursive'
         ).format(self._name)
         reinit_repo = (
             'cd ./{} '
             '&& git remote remove origin '
-            '&& git commit --amend --reset-author -m "Initial commit" > /dev/null 2>&1'
```
Comment (tyoungNIO) [inline nio_cli/commands/new.py]: not a valid path in Windows

File: nio_cli/commands/new.py [hunk 2]
```diff
@@ -27,15 +27,15 @@ def run(self):
             else:
                 repo = 'git://github.com/niolabs/{}.git'.format(self._template)
 
-        clone = "git clone --depth=1 {} {}".format(repo, self._name)
```
Comment (tyoungNIO) [inline nio_cli/commands/new.py]: cause  of issue #84

File: tests/test_cli.py [hunk 3]
```diff
@@ -188,19 +189,20 @@ def test_list_command_with_id(self):
...
+        with patch('builtins.print') as mock_print:
             self._main('list', **{
                 "services": False,
                 '--username': 'user',
                 '--password': 'pwd'
             })
             self.assertEqual(len(responses.calls), 1)
-            self.assertEqual(print.call_count, 2)
-            for index, blk in enumerate(blk_response):
-                self.assertEqual(
```
Comment (tyoungNIO) [inline tests/test_cli.py]: jenkins got these responses out of order when running tests

File: nio_cli/commands/new.py [hunk 4]
```diff
@@ -46,7 +46,8 @@ def run(self):
             for file_name in files:
                 if file_name == 'requirements.txt':
                     reqs = os.path.join(root, file_name)
-                    subprocess.call([sys.executable, '-m', 'pip', 'install', '-r', reqs])
```
Comment (tyoungNIO) [inline nio_cli/commands/new.py]: line too long

**Comment by mattdodge** [comment]
This is an interesting discovery. Since we do intend users to clone the project template and then use that to populate a new repo we should not be using the shallow clone. I hadn't thought of that before but it makes sense.

Ideally they don't see the commit history for the project template but that's not a huge deal. I think cloning and then deleting the `.git` folder and making the user do a `git init` makes more sense anyways. I'm good with this PR as a solution for now though.

**Pull request merged**